apps/blockchain/lib/blockchain/blocktree.ex

```diff
@@ -76,4 +77,19 @@ defmodule Blockchain.Blocktree do
 
     %{blocktree | best_block: new_best_block}
   end
+
+  @doc """
+  Returns the best block in a tree, which is either the listed best block,
+  or it's the genesis block, which we create.
+  """
+  @spec get_best_block(t(), Chain.t(), DB.db()) :: {:ok, Block.t()} | {:error, any()}
+  def get_best_block(blocktree, chain, db) do
+    case blocktree.best_block do
```

Comment:
Would this work?

    if is_nil(blocktree.best_block) do
        {:ok, Genesis.create_block(chain, db)}
    else
      Block.get_block(block.block_hash, db)
    end

I've been going with `if` if there are two branches and `case`/`cond` if there are more then two branches. Happy to use `case/cond` over `if` all the time as long as we're consistent.

Reply:
Sure, this came from the original sync script, happy to convert to `if`.